Fix #2516 - Last post not updated when approving the first post using inline mod tools

Shade- opened the pull request:
#2516

JN-Jones commented:
TBH I'm not sure whether this is the right way to fix this. While this will fix the issue itself I feel like we should instead handle updating the lastposter properly in the datahandler which only updates the last poster if both thread and post are visible: https://github.com/mybb/mybb/blob/feature/inc/datahandlers/post.php#L1021
This may likely also apply to deleting/restoring threads so properly updating the last poster on adding a new post (or approving/unapproving/deleting/restoring) feels more like the proper solution to me @mybb/developers

Shade- commented:
I thought the same thing in #2521, but I didn't get any feedback. We shall update the logic here also: https://github.com/mybb/mybb/blob/feature/inc/datahandlers/post.php#L1095 given `update_last_post()` runs inside this conditional.

JN-Jones commented:
@mybb/developers any other opinion here? Would be nice to have it in 1.8.9

Shade- commented:
What. The. Heck. Still learning git, but I guess I messed up again by rebasing my branch in an attempt to update my branch to the latest commit in the feature branch.

Guess I'm going to close this PR and open a new one with the proper changes for #2521.

Cu8eeeR commented:
Yep, would be nice to see this in 1.8.9

Ben-Conway commented:
@Shade- any update on this one? Being a high priority issue we should try to get this in 1.8.9.

Shade- commented:
I've removed the visibility check to ensure threads' lastposter is updated when a post is made even if the thread is not visible.

Also removed get_thread() as it seems redundant to me (already initialized in [L847](https://github.com/mybb/mybb/blob/feature/inc/datahandlers/post.php#L847)).

Stefan-ST commented on inc/datahandlers/post.php:
```diff
@@ -1018,7 +1018,7 @@ function insert_post()
 		}
 
-		if($visible == 1 && $thread['visible'] == 1)
+		if($visible == 1)
```
This appears to be incorrect. If the thread isn't visible the post count shouldn't be changed.

Stefan-ST commented on inc/datahandlers/post.php:
```diff
@@ -1255,6 +1249,13 @@ function insert_post()
+		// Update forum count
+		if($visible == 1) {
+			update_last_post($post['tid']);
+			update_forum_counters($post['fid'], array("posts" => "+1"));
```
Same here... The forum counters shouldn't be changes if the thread isn't visible. Updating the thread's last post is correct though.

Stefan-ST commented on inc/datahandlers/post.php:
```diff
@@ -1229,11 +1228,9 @@ function insert_post()
-
-			// Update forum count
-			update_last_post($post['tid']);
+			
```
Can you remove the tabs from the empty line, please?

Stefan-ST commented on inc/datahandlers/post.php:
```diff
@@ -1255,6 +1252,12 @@ function insert_post()
+		// Update forum count
+		if($visible == 1) {
+			update_last_post($post['tid']);
+			update_forum_lastpost($thread['fid']);
```
It's not necessary to move update_forum_lastpost. It doesn't need to be updated when the thread isn't visible.

Stefan-ST commented on inc/datahandlers/post.php:
```diff
@@ -1255,6 +1252,12 @@ function insert_post()
 		}
+		
+		// Update forum count
+		if($visible == 1) {
```
Braces should be in it's own line.
https://docs.mybb.com/1.6/Coding-Standards/

Ben-Conway commented:
@Stefan-ST Are you happy with the changes?

Ben-Conway commented:
@Stefan-ST Are these changes okay?

Cu8eeeR commented:
It would be nice to see this in 1.8.10 release

WildcardSearch commented:
I've tested everything seems fine.

@Stefan-ST can you approve the changes?

Ben-Conway commented:
@WildcardSearch as long as you are 100% sure this works correctly we can approve the changes and merge this.

WildcardSearch commented:
@Ben-Conway I'd feel more comfortable waiting on @Stefan-ST or at least have another @mybb/developers or @mybb/sqa check it.

Ben-Conway commented:
@andrewjs18 can you check this?

euantorano commented:
@Stefan-ST Can you please double check the comments regarding your review and either mark it as resolved or expand on changes you wish to see?

Ben-Conway commented:
Thanks @Stefan-ST. @euantorano merging since @WildcardSearch tested it already
